### ISSUE TITLE
Async Daemon Does Not Run When Eager Initializing Store

### ISSUE THREAD

**Issue by kluhman** [JasperFx/marten#2185]
Marten Version: 5.2.0

When using the `InitializeStore()` functionality to eagerly build the document store, the async daemon will not run.

```csharp
services.AddMarten(options =>
{
    ...
}).AddAsyncDaemon(DaemonMode.Solo).InitializeStore();
``` 

It looks like using `InitializeStore` uses the in-memory `StoreOptions` object built during the `AddMarten` call, where as the async daemon uses an object built using `services.Configure`. So calling `InitializeStore` builds and caches the store using options with `AsyncMode = Disabled`, and when the daemon starts, it fails to run. Removing the `InitializeStore()` call fixes the issue.

I'll try to create a PR to fix this tonight

**Comment by jeremydmiller** [comment]
Yeah, don't do that. I really wanted to rip out the InitializeStore() in 5, but forgot about it. It's non-idiomatic .Net usage, and is gonna cause all kinds of hell now. Why do you even need that?

**Comment by kluhman** [comment]
@jeremydmiller at one point I needed it because lazily initializing the store created extra db connections inside an ambient transaction that caused me grief with postgres.

Luckily I don't need that anymore, so I worked around it by just removing the eager initialization in my application. I just figured that the daemon should still work with any valid configuration. Maybe an `Obsolete` attribute on the `InitializeStore` method until v6 will help move people off this method if it won't be supported
